### PR TITLE
Docs: sync plugin JSON info in the different files (first part)

### PR DESCRIPTION
**Pull request by andresmgot**
**What is this feature?**

My goal here is:

 - Check plugin.json docs and see if the descriptions are still accurate or if they need more detail.
 - Check that plugindef.cue and plugin.schema.json are in sync. Having the same description and options, in the same order. This will help with eventual automation.

So far, I have only reviewed the mandatory fields and `$schema` but rather than keep adding things to a big PR, it's better to review these little by little.

**Why do we need this feature?**

[Add a description of the problem the feature is trying to solve.]

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Ref #63614

**Special notes for your reviewer**:

### CODE REVIEW

File: docs/sources/developers/plugins/metadata.md [hunk 1]
```diff
@@ -16,12 +16,12 @@ The plugin.json file is required for all plugins. When Grafana starts, it scans
...
+| `dependencies`       | [object](#dependencies)       | **Yes**  | Dependency information related to Grafana and other plugins.                                                                                                                                                                                                                                                                                                                                            |
+| `$schema`            | string                        | No       | Schema definition for the plugin.json file. Used to verify the validy it.                                                                                                                                                                                                                                                                                                                               |
```
Comment (wbrowne) [inline docs/sources/developers/plugins/metadata.md:24]:
```suggestion
| `$schema`            | string                        | No       | Schema definition for the plugin.json file. Used primarily for schema validation.                                                                                                                                                                                                                                                                                                                           |
```

**Review by wbrowne — APPROVED** [review]

File: docs/sources/developers/plugins/metadata.md [hunk 2]
```diff
@@ -16,12 +16,12 @@ The plugin.json file is required for all plugins. When Grafana starts, it scans
...
-| `info`               | [object](#info)               | **Yes**  | Metadata for the plugin. Some fields are used on the plugins page in Grafana and others on grafana.com if the plugin is published.                                                                                                                                                                                                                                                                      |
+| `id`                 | string                        | **Yes**  | Unique name of the plugin. If the plugin is published on grafana.com, then the plugin id should follow the Grafana naming convention.                                                                                                                                                                                                                                                                   |
```
Comment (josmperez) [inline docs/sources/developers/plugins/metadata.md:19]: change id to ID or `id`

File: pkg/plugins/plugindef/plugindef.cue [hunk 3]
```diff
@@ -31,9 +35,116 @@ seqs: [
 				// a superset of plugin types.
 				#IncludeType: type | "dashboard" | "page"
 
-				// Human-readable name of the plugin that is shown to the user in
-				// the UI.
-				name: string
+				// Metadata about the plugin.
```
Comment (josmperez) [inline pkg/plugins/plugindef/plugindef.cue:38]: Style nit: In blocks below, periods not needed at line ends unless they close a complete sentence.

**Review by josmperez — CHANGES_REQUESTED** [review]
Style suggestions.

**Review by josmperez — APPROVED** [review]

**Pull request merged**